src/utils/response/responseHandlers.ts

```diff
@@ -26,11 +39,15 @@ export async function toApiResponse<T>(promise: RequestPromise<T>): Promise<ApiR
     }
   }
 
-  const apiResponse = resolved.data;
-
+  const apiResponse:
+    | ApiResponseOnFailureServerCommunication
+    | ApiResponseOnSuccess<ResourceName>
+    | ApiResponseOnFailure
+    | ApiResponseOnSuccess<ResourceName, ServerCommunication> = resolved.data;
   // Workaround for serverCommunication endpoint returning Success: true despite an error
+  // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
```

Comment:
You could get rid of this by including `ApiResponseOnFailureServerCommunication` in the `ApiResponse` union, then you can check if `Results === null` without error

Reply:
I actually tried that, but it didn't work.